### ISSUE TITLE
Stubs less accurate than Doctrine Psalm phpdoc

### ISSUE THREAD

**Issue by noemi-salaun** [phpstan/phpstan-doctrine#389]
With the addition of the list type support, I got false positive with `EntityRepository::findBy()` or `findAll()` not returning a list. When looking at the Doctrine class :

```php
    /**
     * Finds entities by a set of criteria.
     *
     * @param int|null $limit
     * @param int|null $offset
     * @psalm-param array<string, mixed> $criteria
     * @psalm-param array<string, string>|null $orderBy
     *
     * @return object[] The objects.
     * @psalm-return list<T>
     */
    public function findBy(array $criteria, ?array $orderBy = null, $limit = null, $offset = null) { }
```

it looks more accurate than the stubs :

```php
	/**
	 * @phpstan-param array<string, mixed> $criteria
	 * @phpstan-param array<string, string>|null $orderBy
	 * @phpstan-param int|null $limit
	 * @phpstan-param int|null $offset
	 * @phpstan-return TEntityClass[]
	 */
	public function findBy(array $criteria, ?array $orderBy = null, $limit = null, $offset = null);
```

Is it better to update the stubs or delete them to use the base phpdoc?

**Comment by ondrejmirtes** [comment]
We can't delete them for users of old Doctrine versions which do not have these PHPDocs.

**Comment by ondrejmirtes** [comment]
Your issue might have been solved: https://github.com/phpstan/phpstan-doctrine/pull/388
